parser.go

```diff
@@ -12,29 +13,34 @@ import (
 )
 
 var (
-	DefaultLexerRules = []lexer.Rule{
-		{"DateTime", `\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(\.\d+)?(-\d\d:\d\d)?`, nil},
-		{"Date", `\d\d\d\d-\d\d-\d\d`, nil},
-		{"Time", `\d\d:\d\d:\d\d(\.\d+)?`, nil},
-		{"TimeFormat", `((%([aAbBhcCdeDHIjmMnprRStTUwWxXyYLz]|E[cCxXyY]|O[deHImMSUwWy]))(\:|\/| |\-|T|Z|\.|))+`, nil},
-		{"Ident", `[a-zA-Z_\.\*\-0-9\/\{\}]+`, nil},
-		{"String", `[a-zA-Z0-9_\.\/\*\-]+`, nil},
-		{"Number", `[-+]?[.0-9]+\b`, nil},
-		{`Float`, `(\d*)(\.)*(\d+)+`, nil},
-		{"Punct", `\[|]|[-!()+/*=,]`, nil},
-		{"Regex", `\^[^\n]+\$`, nil},
-		{"Topic", `\$[a-zA-Z0-9_\.\/\*\-]+`, nil},
-		{"comment", `#[^\n]+`, nil},
-		{"whitespace", `[\ \t]+`, nil},
-		{"EOL", "[\r\n]+", nil},
+	DefaultLexerRules = []lexer.SimpleRule{
+		{"TemplateVariable", `\{\{[^\}]+\}\}`},
+		{"JsonObject", `\{[^\}]+\}`},
+		{"DateTime", `\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d+)?`},
+		{"Date", `\d\d\d\d-\d\d-\d\d`},
+		{"Time", `\d\d:\d\d:\d\d(\.\d+)?`},
+		{"TimeFormat", `((%([aAbBhcCdeDHIjmMnprRStTUwWxXyYLz]|E[cCxXyY]|O[deHImMSUwWy]))(\:|\/| |\-|T|Z|\.|))+`},
+		{"Address", `\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}`},
+		{"Unit", `[-+]?(\d*\.)?\d+([kKmMgG])`},
+		{"Float", `[-+]?(\d*\.)?\d+`},
```

Comment:
point is optional so it will always match integers as floats.
```suggestion
		{"Float", `[-+]?\d*\.\d+`},
```